### PR TITLE
custom reagent pie smite

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

adds a custom reagent pie smite for admins

## Why It's Good For The Game

was asked to do this by augments

## Changelog
:cl:
add: custom reagent pie smite
/:cl:

### CODE REVIEW

**Comment by BlackMajor** [comment]
Blessed content.
Albeit, I question why augments wanted this, seeing as there's already a verb for adding reagents to /atom's

File: code/modules/admin/verbs/randomverbs.dm [hunk 1]
```diff
@@ -1314,6 +1314,36 @@ GLOBAL_LIST_EMPTY(custom_outfits) //Admin created outfits
 		if(ADMIN_PUNISHMENT_PIE)
 			var/obj/item/reagent_containers/food/snacks/pie/cream/nostun/creamy = new(get_turf(target))
 			creamy.splat(target)
+		if (ADMIN_PUNISHMENT_CUSTOM_PIE)
+			var/obj/item/reagent_containers/food/snacks/pie/cream/nostun/A = new(get_turf(target))
+			if(!A.reagents)
+				var/amount = input(usr, "Specify the reagent size of [A]", "Set Reagent Size", 50) as num
```
Comment (Ghommie) [inline code/modules/admin/verbs/randomverbs.dm:1320]: I see all of this (up to add_reagent()) is quite the copypasta of https://github.com/Citadel-Station-13/Citadel-Station-13/blob/8c50b578b4e73e1b1b93e7f5359561aeac899a27/code/datums/datumvars.dm#L944

Could you please move both into a new proc for better maintanance?

File: code/modules/admin/verbs/randomverbs.dm [hunk 2]
```diff
@@ -1314,6 +1314,19 @@ GLOBAL_LIST_EMPTY(custom_outfits) //Admin created outfits
...
```
Comment (Ghommie) [inline code/modules/admin/verbs/randomverbs.dm:1320]: Oh yea, pies have a reagent holder already, this bit is useless. What you may want is to make a subtype that starts off with no reagent inside.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/admin/verbs/randomverbs.dm (hunk 2 above)
Comment (Seris02) [reply code/modules/admin/verbs/randomverbs.dm:1320]: nah, I'd have to rework the initialize on reagent_containters, so nope, not doin that

File: code/__HELPERS/reagents.dm [hunk 3]
```diff
@@ -72,3 +72,24 @@
 	if(!GLOB.chemical_reactions_list[primary_reagent])
 		GLOB.chemical_reactions_list[primary_reagent] = list()
 	GLOB.chemical_reactions_list[primary_reagent] += R
+
+/proc/choose_reagent_id(mob/user)
+	var/chosen_id
+	switch(alert(user, "Choose a method.", "Add Reagents", "Search", "Choose from a list"))
```
Comment (Ghommie) [inline code/__HELPERS/reagents.dm]: You just removed the "I'm feeling lucky" option.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/__HELPERS/reagents.dm (hunk 3 above)
Comment (Seris02) [reply code/__HELPERS/reagents.dm]: fuck you're right

**Review by Ghommie — APPROVED** [review]

**Pull request merged**